pkg/proctree/fileinfo.go

```diff
@@ -0,0 +1,190 @@
+package proctree
+
+import (
+	"sync"
+	"time"
+)
+
+//
+// File Info
+//
+
+// FileInfo represents a file.
+type FileInfo struct {
+	name      string
```

Comment:
I don't really understand what is a file name.
In the last commit you put here the task name, which is not related to the file necessarily.
Maybe export it to the task info?

Reply:
FileInfo describes program executables or interpreters and comes from the following eBPF code:
 
```C
    struct file *file = get_file_ptr_from_bprm(bprm);
    void *file_path = get_path_str(__builtin_preserve_access_index(&file->f_path));

    const char *filename = get_binprm_filename(bprm);
    dev_t s_dev = get_dev_from_file(file);
    unsigned long inode_nr = get_inode_nr_from_file(file);
    u64 ctime = get_ctime_nanosec_from_file(file);
    umode_t inode_mode = get_inode_mode_from_file(file);

    save_str_to_buf(&signal->args_buf, (void *) filename, 1);                   // cmdpath
    save_str_to_buf(&signal->args_buf, file_path, 2);                           // pathname
    save_to_submit_buf(&signal->args_buf, &s_dev, sizeof(dev_t), 3);            // dev
    save_to_submit_buf(&signal->args_buf, &inode_nr, sizeof(unsigned long), 4); // inode
    save_to_submit_buf(&signal->args_buf, &ctime, sizeof(u64), 5);              // ctime
    save_to_submit_buf(&signal->args_buf, &inode_mode, sizeof(umode_t), 6);     // inode_mode

    // The proc_info interpreter field is set by "load_elf_phdrs" kprobe program.
    save_str_to_buf(&signal->args_buf, &proc_info->interpreter.pathname, 7);                           // interpreter_pathname
    save_to_submit_buf(&signal->args_buf, &proc_info->interpreter.id.device, sizeof(dev_t), 8);        // interpreter_dev
    save_to_submit_buf(&signal->args_buf, &proc_info->interpreter.id.inode, sizeof(unsigned long), 9); // interpreter_inode
    save_to_submit_buf(&signal->args_buf, &proc_info->interpreter.id.ctime, sizeof(u64), 10);          // interpreter_ctime
``` 

So you are right, name is confusing. It would be "filename" (from the original existing eBPF code". But then we also have "pathName".

Values from real world:

![image](https://github.com/aquasecurity/tracee/assets/7395852/95b5e28f-2c78-4b9b-8a04-722cdc973603)

Note that the pathName is important specially when there is a symlink. Example: `/usr/bin/dash` working as `/bin/sh`.

Reply:
I'll wait for suggestions here (yours) about the names, its more meaningful to you than to me.

Reply:
In my opinion, there is no need for both of them in the process tree.
The name here is relevant only upon execution, so it is not part of the file resource data.
If someone wants this path they can use the arguments of the process to get it.
So, I think we should keep only the path member.

Reply:
I added comm to TaskInfo and kept both, the file and path into FileInfo until we fully understand implications of having (or not) both, we can always remove it later.